packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -95,72 +100,84 @@ export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps
-          truncatedTitleSecondPiece: originalTitle.slice(originalTitle.length - (maxLength / 2 - TRUNCATION_FIRST_PIECE_LONGER_BY))
-        });
-      } else {
-        // The text is not so long, so we'll just break it into two pieces
-        this.setState({
-          truncatedTitleFirstPiece: originalTitle.slice(0, Math.ceil(originalTitle.length / 2) + TRUNCATION_FIRST_PIECE_LONGER_BY),
-          truncatedTitleSecondPiece: originalTitle.slice(
-            originalTitle.length - Math.floor(originalTitle.length / 2) + TRUNCATION_FIRST_PIECE_LONGER_BY
-          )
-        });
-      }
+  // Truncate logic here way can't handle the case that chars with different widths are mixed very well, let _shrinkTitle take care of that.
+  private _truncateTitle = (): void => {
+    if (!this.state.needMeasurement) {
+      return;
     }
 
-    // Save the width we just started truncation at, so that later we will only update truncation if necessary
-    if (this._titleElement.current) {
-      this._truncatedTitleAtWidth = this._titleElement.current.clientWidth;
+    this._async.requestAnimationFrame(this._truncateWhenInAnimation);
+  };
+
+  private _truncateWhenInAnimation: () => void = () => {
+    const originalTitle = this.props.title;
+    const element: HTMLDivElement | null = this._measureTitleElement.current;
+
+    if (element) {
+      const style: CSSStyleDeclaration = getComputedStyle(element);
+      if (style.width && style.lineHeight && style.height) {
+        const { clientWidth, scrollWidth } = element;
+        const lines: number = Math.floor(
+          (parseInt(style.height, 10) + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD) / parseInt(style.lineHeight, 10)
+        );
+        // Used to calculate the number of chars that should be truncated.
+        const overFlowRate: number = scrollWidth / (parseInt(style.width, 10) * lines);
+
+        if (overFlowRate > 1) {
+          const truncatedLength: number = originalTitle.length / overFlowRate - 3 /** Saved for separator */;
+          return this.setState({
+            truncatedTitleFirstPiece: originalTitle.slice(0, truncatedLength / 2),
+            truncatedTitleSecondPiece: originalTitle.slice(originalTitle.length - truncatedLength / 2),
+            clientWidth,
+            needMeasurement: false
+          });
+        }
+      }
     }
+
+    return this.setState({ needMeasurement: false });
   };
 
-  private _shrinkTitle(): void {
-    if (this._doesTitleOverflow()) {
-      const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
-      this._isTruncated = true;
+  private _shrinkTitle: () => void = () => {
+    const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
+    if (truncatedTitleFirstPiece && truncatedTitleSecondPiece) {
+      const titleElement = this._titleElement.current;
+
+      if (!titleElement) {
+        return false;
```

Comment:
Why is the return type void if you return false here?

Reply:
Addressed, thanks